src/main/java/com/slack/kaldb/server/KaldbQueryService.java

```diff
@@ -15,26 +25,73 @@
 
   public static List<String> servers = new ArrayList<>();
 
-  /**
-   * TODO 1. Cache the stub 2. Run in parallel 3. Integrate with ZK to update list 4. Aggregate the
-   * responses instead of only returning the result from the first server
-   */
-  public KaldbSearch.SearchResult distributedSearch(KaldbSearch.SearchRequest request) {
+  // public so that we can override in tests
+  public static int READ_TIMEOUT_MS = 15000;
```

Comment:
This should probably be a configuration option?

Reply:
good call! Will fix

Reply:
+1 to config option. Also, let's change the default time to 30 seconds or so for a start since we don't want to timeout queries too early until the system is optimal.